colossalai/fx/profiler/profiler.py

```diff
@@ -279,31 +294,38 @@ def profile_function(target: 'Target', device: str = 'meta') -> Callable:
 
     def f(*args: Tuple[Argument, ...], **kwargs: Dict[str, Any]) -> Any:
 
+        # find the grad for parameter in args and kwargs
+        param_size = 0
+
+        def get_param_size(x):
+            nonlocal param_size
+            if isinstance(x, Parameter):
+                param_size += activation_size(x)
+
+        tree_map(get_param_size, args)
+        tree_map(get_param_size, kwargs)
+
         # If there is an argument that this `call_function` is inplace, we should
         # still run the profiling but discard some results regarding `target`
+        global do_not_cache
```

Comment:
This happens if we replace `inplace=True` with `inplace=False`, because some tensors might be saved for backward during `inplace=False`, and it will be cached.